spec/loan_creator/common_spec.rb

```diff
@@ -0,0 +1,93 @@
+# coding: utf-8
+require 'spec_helper'
+
+describe LoanCreator::Common do
+  describe '#leap_days_count' do
+    subject {
+      Class.new(described_class) {
+        def initialize; end
+      }.new.send(:leap_days_count, start_date, end_date)
```

Comment:
Définir `def initialize; end` alors que tout ce qu'on veut est tester `leap_days_count` me parait étrange, et cela me fait dire que `leap_days_count`, n'utilisant aucun attribut ni méthode de l'instance de `LoanCreator::Common`, n'a pas besoin d'être une méthode d'instance de celle-ci mais devrait plutôt être une méthode de classe, voire d'un module séparé.

Les méthodes `compute_realistic_periodic_interests_rate` et `multi_part_interests` semblent aussi assez proches de ce cas.

Les extraire pourrait peut être alléger les responsabilités de `LoanCreator::Common` ?

Reply:
@Kernael @RobertAudi la formation SOLID était sympa en théorie, mais en pratique là j'ai du mal à voir ce qui serait une bonne solution. Vous avez une idée de refactor ?

J'ai l'impression que faire un module ici ne ferait que déplacer la logique sans apporter grand-chose.

Reply:
Sorry impossible de voir que j'avais été taggé ici. D'ailleurs si vous avez un trick pour qu'une PR où quelqu'un te @mention puisse s'afficher dans la liste des PR je suis preneur. (`mentions:Kernael` ne fonctionne pas).

=================


Le helper qu'à fait @Mziserman dans sa dernière itération me semble très bien